drivers/ptp_clock/ptp_clock_ha1588.c

```diff
@@ -0,0 +1,631 @@
+/*
+ * Copyright 2025 CISPA Helmholtz Center for Information Security gGmbH
+ *
+ * Based on ha1588 Linux driver Copyright (C) 2018 Geon Technologies, LLC.
```

Comment:
This is an issue @WorldofJARcraft. The Linux driver is assumedly GPLv2 licensed? That license is not compatible with Apache v2, and you cannot just take the code and "relicense" it here.

Reply:
Maybe the term "based on" is too much.
This driver is a from-scratch implementation by me that I tried to keep device-tree compatible with the Linux driver.
For reference, the Linux driver is here: https://github.com/Geontech/meta-ha1588/blob/master/recipes-kernel/linux/files/ha1588/0002-ha1588-c.patch

Reply:
I see, perhaps you want to rephrase that then (IANAL).

Reply:
I will call it "compatible with" then.